### PR TITLE
coot/ouroboros network 0.3.0.1

### PR DESCRIPTION
**Pull request by coot**
- Require ouroboros-network-0.3.0.1
- Updated cardano-node changelog file for 1.35.6 release

### CODE REVIEW

File: cardano-node/ChangeLog.md [hunk 1]
```diff
@@ -25,13 +25,30 @@ None
   format will be supported for next two major releases of the node (the last
   major version which will support it is `1.37`). (#4563)
 
+- Fixed a tight loop when demoting a peer from hot to warm timeouts
+  (input-output-hk/ouroboros-network#4357).
+
 [#4563]: https://github.com/input-output-hk/cardano-node/issues/4563
 [understanding-config-files]: https://github.com/input-output-hk/cardano-node/blob/master/doc/getting-started/understanding-config-files.md
 
+## 1.35.5 -- January 2023
```
Comment (Jimbo4350) [inline cardano-node/ChangeLog.md:34]: https://github.com/input-output-hk/cardano-node/releases/tag/1.35.5 
Can you ask somebody for the ledger team to update the change log and point to the PR that "addressed a fix in ledger internal data structures."

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

File: cardano-node/ChangeLog.md [hunk 2]
```diff
@@ -25,13 +25,30 @@ None
   format will be supported for next two major releases of the node (the last
   major version which will support it is `1.37`). (#4563)
 
+- Fixed a tight loop when demoting a peer from hot to warm timeouts
```
Comment (Jimbo4350) [inline cardano-node/ChangeLog.md:28]: Why are the changes now part of the 1.35.6 release? They were previously part of the 1.35.5 release, not so?

Comment (coot) [reply cardano-node/ChangeLog.md:28]: These changes were merged after `1.35.5` was released (https://github.com/input-output-hk/cardano-node/commits/release/1.35, `1.35.5` was tagged at 8762a10ef).

**Review by Jimbo4350 — APPROVED** [review]

**Comment by coot** [comment]
Thanks @angerman 

bors merge

**Comment by coot** [comment]
bors merge

**Comment by jbgi** [comment]
bors retry

**Pull request merged**